### ISSUE TITLE
Save on focus out?

### ISSUE THREAD

**Issue by sinegovsky-ivan** [qontu/ngx-inline-editor#21]
Hi,

Is any way to make input changes after "focus out" ?

**Comment by Caballerog** [comment]
Hi @sinegovsky-ivan if you use `[(ngModel)]='myVar' `in the ng2-inline-editor you could change the value from your TS.

TS
`
myVar = 'newValue';
`

**Comment by sinegovsky-ivan** [comment]
Your not understood me.

I want remove buttons ("save" and "cancel") and change behavior to "text saved after input loses focus"

**Comment by Caballerog** [comment]
Hi @sinegovsky-ivan,

I understand you now! I believe that I should implement @Output onFocus to implement this feature.

At the moment, I think that is impossible.

I mark this feature like important too!

**Comment by sinegovsky-ivan** [comment]
Thank you!

**Comment by KeithGillette** [comment]
It would be great if the component API allowed access to the "save" and "cancel" button methods generically in such a way as we could bind arbitrary events to trigger them. For example:

```html
<inline-editor 
  (keyup.enter)="onSubmit(value)"
  (keyup.esc)="cancel(value)"
  type="text" [(ngModel)]="editableText" (onSave)="saveEditable($event)" name="editableText1" 
  >
</inline-editor>
```

**Comment by twmiller** [comment]
Yeah, I think that @KeithGillette is onto something I need.  I'd like to have the return/enter key be an option for saving a field, rather than having to have my users click the check button after every field edit.

**Comment by MaZZly** [comment]
This is something I'd like as well. both being able to bind key events to save/cancel as well as having focusOut perform save

**Comment by xxxtonixxx** [comment]
@KeithGillette @twmiller @MaZZly thanks for your feedback, can you open other issue to track that feature? 👍

**Comment by KeithGillette** [comment]
Enhancement request #47 opened, @xxxtonixxx. Thanks!

**Comment by xxxtonixxx** [comment]
@sinegovsky-ivan, @twmiller  follow #57, when that PR be merged, you will be able to use `saveOnBlur` property to save on focus out and `hideButtons` to hide buttons. You can see #30 and #47, there are examples

**Comment by xxxtonixxx** [comment]
This should be fixed in the last version (`0.2.0-alpha.0`). With the change of name, you should do an `npm i @qontu/ngx-inline-editor` and update your imports 👍
